scalameter-core/src/main/scala/org/scalameter/execution/invocation/InvocationCountMatcher.scala

```diff
@@ -0,0 +1,104 @@
+package org.scalameter.execution.invocation
+
+import java.lang.reflect.Method
+import java.util.regex.Pattern
+import org.objectweb.asm.Type
+import scala.util.matching.Regex
+import InvocationCountMatcher._
+
+
+/** Object that matches the methods whose invocations should be counted.
+ *
+ *  @param classMatcher matches full class name given in internal format ('/' instead of a '.' as a separator)
+ *  @param methodMatcher matches a specific method
+ */
+case class InvocationCountMatcher(classMatcher: ClassMatcher, methodMatcher: MethodMatcher) {
+  def classMatches(className: String): Boolean = classMatcher.matches(className)
+
+  def methodMatches(methodName: String, methodDescriptor: String): Boolean = methodMatcher.matches(methodName, methodDescriptor)
+}
+
+object InvocationCountMatcher {
+  /** Mixin used for selecting classes whose methods will be checked against [[MethodMatcher]] and if matched,
+   *  counted by a method invocation counting measurer.
+   */
+  sealed trait ClassMatcher {
+    /** Matches class name given either in a normal format or internal jvm format ('/' instead of a '.').
```

Comment:
Sorry if this scaladoc comment is misleading.
It actually means that `matches` method will be invoked with a class name in internal format (with '/') since it will be called from instrumentation code.
On the other hand, I didn't want to bother user to use '/' instead of a '.', so that's the reason I've chosen to replace every '/' with a '.' in `className` argument in every `matches` implementation.